source/lac/sparsity_tools.cc

```diff
@@ -925,21 +925,19 @@ namespace SparsityTools
     for (DynamicSparsityPattern::size_type i = 0; i < rows_per_cpu.size(); ++i)
       start_index[i + 1] = start_index[i] + rows_per_cpu[i];
 
-    using map_vec_t = std::map<DynamicSparsityPattern::size_type,
-                               std::vector<DynamicSparsityPattern::size_type>>;
+    using map_vec_t =
+      std::map<unsigned int, std::vector<DynamicSparsityPattern::size_type>>;
 
     map_vec_t send_data;
 
     {
-      unsigned int dest_cpu = 0;
-
-      DynamicSparsityPattern::size_type n_local_rel_rows = myrange.n_elements();
+      unsigned int dest_cpu         = 0;
+      const auto   n_local_rel_rows = myrange.n_elements();
       for (DynamicSparsityPattern::size_type row_idx = 0;
            row_idx < n_local_rel_rows;
            ++row_idx)
```

Comment:
Can't this be a range-based for loop?

Reply:
> Can't this be a range-based for loop?

not with the current code, which does `row_idx += rows_per_cpu[myid] - 1;` below.

Reply:
but let me re-write it a bit...

Reply:
Do it if you feel like it. It's really not crucial.

Reply:
I will wait for tester to be done and push update...

Reply:
The tester is done. :wink:

Reply:
@masterleinad I pushed commit which turns this into a range loop.